File: python/docs/examples/mobile-use/mobile_get_adb_url_example.py
"""
Example: Get ADB Connection URL for Mobile Device

This example demonstrates how to use session.mobile.get_adb_url() to retrieve
an ADB connection URL for connecting to a mobile device. This API is only
available in mobile environments using the mobile_latest image.

The get_adb_url method requires an ADB public key for authentication and
returns the ADB connection URL that can be used with the `adb connect` command.
"""

import os
from agentbay import AgentBay
from agentbay.session_params import CreateSessionParams


def main():
    # Get API key from environment variable
    api_key = os.environ.get("AGENTBAY_API_KEY")
    if not api_key:
        raise ValueError("AGENTBAY_API_KEY environment variable is not set")

    # Initialize AgentBay client
    agent_bay = AgentBay(api_key=api_key)

    # Create a mobile session
    # NOTE: This will only work with mobile_latest image
    print("Creating a mobile session...")
    params = CreateSessionParams(image_id="mobile_latest")
    create_result = agent_bay.create(params=params)

    if not create_result.success:
        print(f"Failed to create session: {create_result.error_message}")
        return

    session = create_result.session
    print(f"✅ Session created: {session.session_id}")
    print(f"   Request ID: {create_result.request_id}")

    try:
        # Example ADB public key
        # In a real scenario, this would come from your ADB setup
        # Replace this with your actual ADB public key
        adbkey_pub = "QAAAAM0muSn7yQCY...your_adb_public_key...EAAQAA="

        # Get ADB URL for the mobile device
        print("\nRetrieving ADB connection URL...")
        adb_result = session.mobile.get_adb_url(adbkey_pub)

        if adb_result.success:
            print(f"✅ ADB URL retrieved successfully")
            print(f"   URL: {adb_result.data}")
            print(f"   Request ID: {adb_result.request_id}")

            # You can now use this URL to connect via ADB
            print(f"\nYou can use this command to connect:")
            print(f"   {adb_result.data}")
        else:
            print(f"❌ Failed to retrieve ADB URL: {adb_result.error_message}")

    finally:
        # Clean up: delete the session
        print("\nCleaning up...")
        delete_result = agent_bay.delete(session)
        if delete_result.success:
            print(f"✅ Session deleted successfully")
            print(f"   Request ID: {delete_result.request_id}")
        else:
            print(f"❌ Failed to delete session: {delete_result.error_message}")


if __name__ == "__main__":
    main()

File: java/agentbay/src/main/java/com/aliyun/agentbay/examples/MobileGetAdbUrlExample.java
package com.aliyun.agentbay.examples;

import com.aliyun.agentbay.AgentBay;
import com.aliyun.agentbay.model.AdbUrlResult;
import com.aliyun.agentbay.model.DeleteResult;
import com.aliyun.agentbay.model.SessionResult;
import com.aliyun.agentbay.session.CreateSessionParams;
import com.aliyun.agentbay.session.Session;

/**
 * Example: Get ADB Connection URL for Mobile Device
 * 
 * This example demonstrates how to use session.mobile.getAdbUrl() to retrieve
 * an ADB connection URL for connecting to a mobile device. This API is only
 * available in mobile environments using the mobile_latest image.
 * 
 * The getAdbUrl method requires an ADB public key for authentication and
 * returns the ADB connection URL that can be used with the `adb connect` command.
 */
public class MobileGetAdbUrlExample {
    
    public static void main(String[] args) {
        try {
            // Get API key from environment variable
            String apiKey = System.getenv("AGENTBAY_API_KEY");
            if (apiKey == null || apiKey.isEmpty()) {
                throw new IllegalStateException("AGENTBAY_API_KEY environment variable is not set");
            }
            
            // Initialize AgentBay client
            AgentBay agentBay = new AgentBay();
            
            // Create a mobile session
            // NOTE: This will only work with mobile_latest image
            System.out.println("Creating a mobile session...");
            CreateSessionParams params = new CreateSessionParams();
            params.setImageId("mobile_latest");
            
            SessionResult createResult = agentBay.create(params);
            
            if (!createResult.isSuccess()) {
                System.out.println("Failed to create session: " + createResult.getErrorMessage());
                return;
            }
            
            Session session = createResult.getSession();
            System.out.println("✅ Session created: " + session.getSessionId());
            System.out.println("   Request ID: " + createResult.getRequestId());
            
            try {
                // Example ADB public key
                // In a real scenario, this would come from your ADB setup
                // Replace this with your actual ADB public key
                String adbkeyPub = "QAAAAM0muSn7yQCY...your_adb_public_key...EAAQAA=";
                
                // Get ADB URL for the mobile device
                System.out.println("\nRetrieving ADB connection URL...");
                AdbUrlResult adbResult = session.mobile.getAdbUrl(adbkeyPub);
                
                if (adbResult.isSuccess()) {
                    System.out.println("✅ ADB URL retrieved successfully");
                    System.out.println("   URL: " + adbResult.getData());
                    System.out.println("   Request ID: " + adbResult.getRequestId());
                    
                    // You can now use this URL to connect via ADB
                    System.out.println("\nYou can use this command to connect:");
                    System.out.println("   " + adbResult.getData());
                } else {
                    System.out.println("❌ Failed to retrieve ADB URL: " + adbResult.getErrorMessage());
                }
                
            } finally {
                // Clean up: delete the session
                System.out.println("\nCleaning up...");
                DeleteResult deleteResult = session.delete();
                if (deleteResult.isSuccess()) {
                    System.out.println("✅ Session deleted successfully");
                    System.out.println("   Request ID: " + deleteResult.getRequestId());
                } else {
                    System.out.println("❌ Failed to delete session: " + deleteResult.getErrorMessage());
                }
            }
            
        } catch (Exception e) {
            System.err.println("Error occurred: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
